job_creator.php

```diff
@@ -132,10 +132,24 @@ public function getInstallerVersion(
 
     public function createJob(int $phpIndex, array $opts): array
     {
+        $isInstallerModule = in_array($this->repoName, INSTALLER_MODULES);
+        $isPreferLoweset = strpos($opts['composer_args'] ?? '', '--prefer-lowest') !== false;
+        $installerVersion = $this->installerVersion;
+        $cmsMajor = BranchLogic::getCmsMajor($this->repoData, $this->branch, $this->getComposerJsonContent());
+        if ($cmsMajor >= 5 && !$isInstallerModule && $isPreferLoweset && $installerVersion) {
+            // --prefer-lowest jobs should use the lowest installer version for non-installer modules
+            // gha-ci will increment the installer version higher if composer is unable to install
+            // and try installing again
+            //
+            // Was going to use something like 5.2.x-dev, install 5.0.x-dev instead
+            $installerVersion = preg_replace('#^([0-9])\.[0-9]\.x-dev$#', '$1.0.x-dev', $installerVersion);
+            // Was going to use something like 5.x-dev, install 5.0.x-dev instaed
+            $installerVersion = preg_replace('#^([0-9])\.x-dev$#', '$1.0.x-dev', $installerVersion);
```

Comment:
Should detect the lowest supported version based on the composer dependencies! That's what this is all about.

Reply:
The detection is done gha-ci rather than gha-generate-matrix

Makes far more sense to use composer to work out what's installable and what's not rather than trying to work out what's installable and what's not by looking at composer.json files in gha-generate-matrix